Always position the window top-right

jenslind opened the pull request:
Fixes https://github.com/muan/mojibar/issues/18
Requires menubar `v2.1.0` :cat2:

muan commented:
Hai! Thank you so much, though I wonder if you could explain the changes a bit? I prepared a branch after seeing your PR to menubar, https://github.com/muan/mojibar/compare/window, afaik it works and the changes were simple. I'm sure you must have good reasons for the preload window change, most likely I missed something, and so I'm curious what are they, if you would? :blush:

:mountain_railway:

jenslind commented:
Basically the only difference is that I use `menubar.window.show()` instead of emiting the trays `clicked` event. I just prefer it that way. :) Also I think menubar should have something like `menubar.show()` to show the window, then the same events can get triggered as when you click the tray icon.

The `preloadWindow` just creates the window when the app starts so `after-create-window` gets triggered on start instead of when you first click the tray icon.

But yeah, your branch has fewer changes to the code. ;)

muan commented:
> I just prefer it that way. :) 

Perfectly valid. :)

> The `preloadWindow` just creates the window when the app starts so `after-create-window` gets triggered on start instead of when you first click the tray icon.

And how is binding on `after-create-window` better than app `ready`?

jenslind commented:
`mb.window` is not available in the `ready` event, its only available after the window has been created. :)

jenslind commented:
I have opened a [PR](https://github.com/maxogden/menubar/pull/51) that makes `showWindow()` and `hideWindow()` available. This will also fix so the `window` is available after the `ready` event.

jenslind commented:
I have updated the PR to use `showWindow() / hideWindow()`. I also moved the global bind into the ready event. This now requires `2.1.2`.

muan commented:
Sorry for the late reply, I just got around to testing this with multiple displays, and seems like this branch doesn't work with multiple display – all windows are opened in the primary screen. I can take a closer look later.

jenslind commented:
Hmm yes, the problem is that `x` are getting a value on `ready` but it never updates on click or when the global key combo is fired. I tried to set `x` in the `show` event but that didn't really work as the `x` and `y` are set before the event is emitted right now. Not sure if there is a solution right now. But someday `electron-positioner` (https://github.com/maxogden/menubar/pull/52) might be used in menubar. Then this will be done with ease. :cat2: :)

jenslind commented:
Updated this PR to use the latest changes in menubar. This should finally actually work ;)
Also removed `mb.window.setVisibleOnAllWorkspaces(true)` as this is now a default in menubar.

muan commented:
:heart: looks like it works! Thank you so much!!!!! :star2: :star: :star2: :star: :star2: :star:
